carriers/ups.js

```diff
@@ -0,0 +1,243 @@
+
+function confirmUps(trackingNumber) {
+    let sum = 0;
+    for (let index = 2; index <= 16; index++) {
+        var num;
+        const asciiValue = trackingNumber[index].charCodeAt(0);
+        if ((asciiValue >= 48) && (asciiValue <= 57)) {
+            num = parseInt(trackingNumber[index], 10);
+        } else {
+            num = (asciiValue - 63) % 10;
+        }
+
+        if ((index % 2) !== 0) { num = num * 2; }
+        sum += num;
+    }
+
+    const checkdigit = (sum % 10) > 0 ? 10 - (sum % 10) : 0;
+    if (checkdigit === parseInt(trackingNumber[17], 10)) { return [true, true]; }
+    return [false, false];
+}
+
+function confirmUpsFreight(trackingNumber) {
+    const firstChar = `${(trackingNumber.charCodeAt(0) - 63) % 10}`;
+    const remaining = trackingNumber.slice(1);
+    trackingNumber = `${firstChar}${remaining}`;
+    if (checkDigit(trackingNumber, [3, 1, 7], 10)) { return [true, true]; }
+    return [false, false];
+}
+
+function getActivities(package) {
+    var activitiesList = package.Activity;
+
+    if (activitiesList.length != undefined) {
+        activitiesList.forEach(activity => {
+            if (activity.ActivityLocation != undefined) {
+                activity.address = {
+                    city: activity.ActivityLocation.Address === undefined ? (activity.ActivityLocation.City === undefined ? '' : activity.ActivityLocation.City) : (activity.ActivityLocation.Address.City === undefined ? '' : activity.ActivityLocation.Address.City),
+                    state: activity.ActivityLocation.Address === undefined ? (activity.ActivityLocation.StateProvinceCode === undefined ? '' : activity.ActivityLocation.StateProvinceCode) : (activity.ActivityLocation.Address.StateProvinceCode === undefined ? '' : activity.ActivityLocation.Address.StateProvinceCode),
+                    country: activity.ActivityLocation.Address === undefined ? (activity.ActivityLocation.CountryCode === undefined ? '' : activity.ActivityLocation.CountryCode) : (activity.ActivityLocation.Address.CountryCode === undefined ? '' : activity.ActivityLocation.Address.CountryCode),
+                    zipcode: activity.ActivityLocation.Address === undefined ? (activity.ActivityLocation.PostalCode === undefined ? '' : activity.ActivityLocation.PostalCode) : (activity.ActivityLocation.Address.PostalCode === undefined ? '' : activity.ActivityLocation.Address.PostalCode)
+                }
+                activity.location = geography.addressToString(activity.address);
+            } else {
+                activity.address = {
+                    city: '',
```

Comment:
Don't use empty string to represent null or undefined: https://coding.abel.nu/2013/01/on-null-or-why-empty-strings-are-not-same-as-null/